### DISCUSSION TITLE
Companion App - MiniMed Mobile communications?

### DISCUSSION THREAD

**Discussion by TerrysNightscout** [NightscoutFoundation/xDrip#2629] [Q&A] [answered]
When selecting in xDrip Hardware Data Source on a master phone that has MiniMed Mobile app running, how/where does xDrip know that the Medtronic app is the exact source you are trying to import into it?? There could be other apps running (maybe Dexcom), so how do you direct the MiniMed Mobile notifications as the source for xDrip????

**Comment by jamorham** [comment, marked as answer]
It looks for sources that match known apps to retrieve glucose numbers. If you have glucose numbers from two different apps then this would be a very unusual setup as far as I know and xDrip would probably would try to interpret both. If there is an actual problem occurring, please describe it and we can look for resolutions.

**Reply by steve8x8** [reply]
Just an idea: similar to the logging settings, could we have a source selector that limits the set of allowed source apps (to one ;) as one could address a second one by using an xDrip variant)? Default setting: no selection, process everything seen in notifications

**Comment by tim2000s** [comment]
I'm running xDrip (regular) as the collector  with Dexcom ONE. I'm also running G7, using BYODA as the collector. 

xDrip regular has "identify receiver" set to AndroidAPS.

Additionally, I've set up xDrip variant (dated 12th Jan 2023 build) to use "companion app" with the intention of forwarding G7 data to Nightscout to compare the two. 

xDrip variant doesn't receive any data and appears to show that it can't find the Dexcom G7 BYODA app in the logs (attached). When "Companion app Bluetooth" is enabled, it pulls data from the Dexcom ONE. 
![Screenshot_20230116_092133_xDrip+ variant4](https://user-images.githubusercontent.com/14833174/212651553-9e24f432-011f-4607-a3ca-bcde26791bbd.jpg)


Any ideas how to get the G7 data into xDrip Variant?

**Comment by tloczekt** [comment]
Is there a possibility that Minimed Mobile notifications should be only  in English language? (I mean IOB and messages) . I am using polish version of Minimed, one other colleague from France french version?
